src/provisioner/cluster.py

```diff
@@ -65,17 +73,101 @@ def _get_kvs(self, prefix, node):
         elif isinstance(node, list):
             for i, n in enumerate(node):
                 kvs.extend(self._get_kvs(f'{prefix}[{i}]', n))
-        elif isinstance(node, str):
+        elif isinstance(node, str) or type(node) == int:
             kvs.append((prefix, node))
         return kvs
 
     def save(self, config_store):
         """ Saves cluster information onto the conf store """
 
         kvs = []
-        for node in self._node_list:
-            node_id = node['id']
-            key_prefix = f'node>{node_id}'
-            kvs.extend(self._get_kvs(key_prefix, node))
+        try:
+            for node in self._node_list:
+                node_id = node.pop('id')
+                node['storage_set_count'] = 1
```

Comment:
We should not hard code storage_set_count to 1.

Reply:
Added code for calculating  storage_set_count :
```
storage_sets = Conf.get('index', 'cluster>storage_sets')
storage_set_count = len(storage_sets)
```